### ISSUE TITLE
quad_float constructors and casts

### ISSUE THREAD

**Issue by sinandredemption** [BoostGSoC21/multiprecision#63]
This issue tracks the implementation of constructors and casts for `cpp_quad_float<>` class.

**Comment by sinandredemption** [comment]
[Constructors in this commit](https://github.com/BoostGSoC21/multiprecision/blob/b8052d2c0c7e7d5cdda2138f4f1fa6698bdf8052/include/boost/multiprecision/cpp_quad_float.hpp#L159) are basically quad-ified adaptations of `cpp_double_float<>` constructors, as are [signed casts](https://github.com/BoostGSoC21/multiprecision/blob/b8052d2c0c7e7d5cdda2138f4f1fa6698bdf8052/include/boost/multiprecision/cpp_quad_float.hpp#L266).

The unsigned casts, however, are implemented via a cast to signed type, and finally recast to unsigned, [as done here](https://github.com/BoostGSoC21/multiprecision/blob/b8052d2c0c7e7d5cdda2138f4f1fa6698bdf8052/include/boost/multiprecision/cpp_quad_float.hpp#L275). Please do provide feedback if this is something incorrect or if you have suggestions.

**Comment by sinandredemption** [comment]
@ckormanyos Do we need to implement more cast operators (possibly cats to other Boost.Multiprecision) types, or shall we close this issue?

**Comment by ckormanyos** [comment]
> Do we need to implement more cast operators (possibly cats to other Boost.Multiprecision) 

Hi @sinandredemption yes Fahad, feel free to close this issue.

We will be reviewing the constructors and we might need a little help from John on this. I am studying one potential ambiguity when using `float128` in combination with `double`/`long` `double` on GCC within Math's special functions. But for this potential ambiguity (which I have not previously mentioned and is invisible in today's CI) we will handle in a future issue dedicated to it once I get a more clear picture of what's going on.

Long story short, Fahad, please feel free to close this issue. Good job on it as well!
